src/poetry/repositories/abstract_repository.py

```diff
@@ -30,7 +31,7 @@ def search(self, query: str) -> list[Package]:
     @abstractmethod
     def package(
         self,
-        name: str,
+        name: NormalizedName,
```

Comment:
I'm wondering that mypy doesn't complain. There are more subclasses that still take pretty name and iirc some need the pretty name.

Reply:
Could you point out what you mean?  Like mypy, I didn't notice anything (though per the commit it's some weeks since I looked at this and so it's not fresh in my mind).

Reply:
https://github.com/python-poetry/poetry/blob/0c8e5b43d30d86d6ba7dc9b3068db26042e9cc53/src/poetry/repositories/cached_repository.py#L70-L78

https://github.com/python-poetry/poetry/blob/0c8e5b43d30d86d6ba7dc9b3068db26042e9cc53/src/poetry/repositories/legacy_repository.py#L48-L72

https://github.com/python-poetry/poetry/blob/0c8e5b43d30d86d6ba7dc9b3068db26042e9cc53/src/poetry/repositories/repository_pool.py#L110-L127

Reply:
ah, somehow machine and man both missed that the legacy repository uses the un-normalized name: [here](https://github.com/python-poetry/poetry/blob/0c8e5b43d30d86d6ba7dc9b3068db26042e9cc53/src/poetry/repositories/cached_repository.py#L77).

I'll remove this change from the collection